### PR TITLE
Some improvements to job list page speed with many jobs

### PR DESCRIPTION
**Pull request by gschueler**
**Is this a bugfix, or an enhancement? Please describe.**
* job list page with many jobs is slow

**Describe the solution you've implemented**

* reduce inefficient resource access and curtail unnecessary code paths
* fix UI issue when job list paging was enabled

### CODE REVIEW

**Comment by jessemarple** [comment]
> tested local. Looks good.

@jtobard - here's 2K jobs https://github.com/rundeckpro/rundeckpro/files/3984649/many_jobs_2000-20191219-135237.rdproject.jar.zip

Can you add that job and give this a test.

File: rundeckapp/grails-app/views/menu/jobs.gsp [hunk 1]
```diff
@@ -24,7 +24,7 @@
     <meta name="skipPrototypeJs" content="true"/>
     <g:set var="projectName" value="${params.project ?: request.project}"/>
     <g:set var="projectLabel" value="${session.frameworkLabels?session.frameworkLabels[projectName]:projectName}"/>
-    <g:set var="paginateJobs" value="${grailsApplication.config.rundeck.gui.paginatejobs}" />
+    <g:set var="paginateJobs" value="${grailsApplication.config.rundeck.gui.paginatejobs.enabled}" />
```
Comment (sjrd218) [inline rundeckapp/grails-app/views/menu/jobs.gsp]: If this property is set to false in the rundeck-config.properties it seems to still show the pagination controls.

**Comment by ltamaster** [comment]
I tested it, 1200 jobs in 9s more less (like 2 min without the fix)

File: rundeckapp/grails-app/views/menu/jobs.gsp (hunk 1 above)
Comment (gschueler) [reply rundeckapp/grails-app/views/menu/jobs.gsp]: ok i'll check that

Comment (gschueler) [reply rundeckapp/grails-app/views/menu/jobs.gsp]: should be fixed now

**Comment by gschueler** [comment]
ref #1351

**Pull request merged**